lib/api.ex

```diff
@@ -4,22 +4,28 @@ defmodule Onvif.API do
   @spec client(Onvif.Device.t(), Keyword.t()) :: Tesla.Client.t()
   def client(device, opts \\ [service_path: :device_service_path]) do
     adapter = {Tesla.Adapter.Finch, name: Onvif.Finch}
-    service_path = Map.fetch!(device, opts[:service_path])
-    uri = device.address <> service_path
-    parsed_uri = URI.parse(uri)
-    no_userinfo_uri = %URI{parsed_uri | userinfo: nil} |> URI.to_string()
-
-    middleware = [
-      {Tesla.Middleware.BaseUrl, no_userinfo_uri},
-      auth_function(device),
-      {Tesla.Middleware.Logger, log_level: :info},
-      {Tesla.Middleware.Headers,
-       [
-         {"connection", "keep-alive"}
-       ]}
-    ]
-
-    Tesla.client(middleware, adapter)
+
+    case Map.fetch!(device, opts[:service_path]) do
```

Comment:
I'd prefer a dedicated private function here instead of a big case statement. Especially because the failure path is just a raise, it doesn't matter with control flow

```
def get_service_path!(device, opts[:service_path]) do
  case Map.fetch!(device, opts[:service_path]) do
    nil -> raise "The service operation is not supported by the device"
    service_path -> service_path
  end
end
```